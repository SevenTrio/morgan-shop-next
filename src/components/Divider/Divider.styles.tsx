import styled, { css } from 'styled-components';

import { ISpacingProps } from 'types/spacing.types';

import spacingStyles from 'utils/spacingStyles';

interface IStyledDividerProps extends ISpacingProps {
  text?: string;
}

export const StyledDivider = styled.div<IStyledDividerProps>`
  ${spacingStyles};

  position: relative;
  height: 1px;
  width: 100%;
  margin: 0;
  border: none;
  border-bottom: 1px solid ${({ theme }) => theme.palette.divider};

  ${({ text }) =>
    text &&
    css`
      &:after {
        content: '${text}';
        position: absolute;
        top: 50%;
        left: 50%;
        margin-right: -50%;
        padding: 0 8px;
        transform: translate(-50%, -50%);
        z-index: 1;
        font-size: 0.875rem;
        line-height: 1.6;
        color: ${({ theme }) => theme.palette.divider};
        background-color: ${({ theme }) => theme.palette.background.default};

        @media screen and (max-width: ${({ theme }) =>
            theme.breakpoints.sm}px) {
          padding: 0 4px;
          font-size: 0.625rem;
        }
      }
    `}
`;
