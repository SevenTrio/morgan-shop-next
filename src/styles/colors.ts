const colors = {
  black: `#121212`,
  darkGray: `#6D6D6D`,
  gray: `#DEDEDE`,
  lightGray: `#F0F0F0`,
  white: `#FAFAFA`,
  red: `#D92222`,
};

export default colors;
