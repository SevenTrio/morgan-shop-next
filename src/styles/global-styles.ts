import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  html {
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    font-size: 16px;
    font-family: 'Poppins', sans-serif;
  }

  img {
    display: block;
    max-width: 100%;
    overflow: hidden;
  }

  p,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 0;
  }
`;
