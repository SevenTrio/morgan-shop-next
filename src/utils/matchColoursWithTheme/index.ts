import { matchColoursWithTheme } from './matchColoursWithTheme';

export default matchColoursWithTheme;
