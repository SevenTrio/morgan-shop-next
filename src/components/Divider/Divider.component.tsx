import React, { HTMLAttributes } from 'react';
import { ISpacingProps } from '../../types/spacing.types';

import { StyledDivider } from './Divider.styles';

export interface IDividerProps
  extends ISpacingProps,
    HTMLAttributes<HTMLDivElement> {
  /**
   * Divider text.
   */
  text?: string;
}

export const Divider: React.FC<IDividerProps> = ({
  text = ``,
  ...htmlProps
}) => {
  return <StyledDivider text={text} {...htmlProps} />;
};

Divider.displayName = `Divider`;
