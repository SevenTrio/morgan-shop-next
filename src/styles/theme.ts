import { DefaultTheme } from 'styled-components';
import { lighten, darken } from 'polished';

import spacing from 'utils/spacing';
import colors from 'styles/colors';

export const theme: DefaultTheme = {
  breakpoints: {
    xs: 0,
    sm: 600,
    md: 960,
    lg: 1280,
    xl: 1920,
  },
  palette: {
    common: {
      black: colors.black,
      darkGray: colors.darkGray,
      gray: colors.gray,
      lightGray: colors.lightGray,
      white: colors.white,
      red: colors.red,
    },
    primary: {
      light: `#424242`,
      main: colors.black,
      dark: darken(0.08, colors.black),
      contrastText: colors.white,
    },
    secondary: {
      light: lighten(0.08, colors.white),
      main: colors.white,
      dark: darken(0.08, colors.white),
      contrastText: colors.black,
    },
    error: {
      light: lighten(0.08, colors.red),
      main: colors.red,
      dark: darken(0.08, colors.red),
      contrastText: colors.white,
    },
    text: {
      primary: colors.black,
      secondary: colors.darkGray,
      disabled: colors.darkGray,
      hint: colors.darkGray,
    },
    divider: colors.gray,
    background: {
      default: colors.white,
      productCard: colors.lightGray,
      menu: colors.lightGray,
    },
  },
  shadows: [`1px 1px 10px rgba(0, 0, 0, 0.08)`],
  spacing,
  zIndexes: {
    basket: 100,
  },
};
