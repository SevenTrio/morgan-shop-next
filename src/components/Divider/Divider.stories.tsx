import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Divider } from './Divider.component';

export default {
  title: `Components/Divider`,
  component: Divider,
} as ComponentMeta<typeof Divider>;

const Template: ComponentStory<typeof Divider> = (args) => (
  <Divider {...args} />
);

export const Default = Template.bind({});
Default.args = {
  text: ``,
};

export const WithText = Template.bind({});
WithText.args = {
  text: `or`,
};
