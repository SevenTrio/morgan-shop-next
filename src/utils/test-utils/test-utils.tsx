import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';

import { GlobalStyle } from 'styles/global-styles';
import { theme } from 'styles/theme';

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Normalize />
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
