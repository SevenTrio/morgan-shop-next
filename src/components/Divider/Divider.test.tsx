import { render } from 'utils/test-utils';
import { theme } from 'styles/theme';

import Divider from 'components/Divider';

const dividerColor = theme.palette.divider;

describe(`Divider component`, () => {
  describe(`renders correctly`, () => {
    test(`without props`, () => {
      const { container } = render(<Divider />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
    });

    test(`with text prop`, () => {
      const dividerText = `Lorem ipsum dolor.`;
      const { container } = render(<Divider text={dividerText} />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`applies correct styles`, () => {
    test(`by default`, () => {
      const { container } = render(<Divider />);
      const tree = container.firstChild;

      expect(tree).toHaveStyleRule(
        `border-bottom`,
        `1px solid ${dividerColor}`,
      );
    });
  });
});
