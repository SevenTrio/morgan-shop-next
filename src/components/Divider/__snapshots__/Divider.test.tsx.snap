// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component renders correctly with text prop 1`] = `
.c0 {
  position: relative;
  height: 1px;
  width: 100%;
  margin: 0;
  border: none;
  border-bottom: 1px solid #DEDEDE;
}

.c0:after {
  content: 'Lorem ipsum dolor.';
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  padding: 0 8px;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 1;
  font-size: 0.875rem;
  line-height: 1.6;
  color: #DEDEDE;
  background-color: #FAFAFA;
}

@media screen and (max-width:600px) {
  .c0:after {
    padding: 0 4px;
    font-size: 0.625rem;
  }
}

<div
  class="c0"
/>
`;

exports[`Divider component renders correctly without props 1`] = `
.c0 {
  position: relative;
  height: 1px;
  width: 100%;
  margin: 0;
  border: none;
  border-bottom: 1px solid #DEDEDE;
}

<div
  class="c0"
/>
`;
